components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/internal/handler/UserOperationEventHookHandler.java

```diff
@@ -223,8 +227,14 @@ public void handleEvent(Event event) throws IdentityEventException {
                             .buildSecurityEventToken(eventPayload, eventUri);
                     EventHookHandlerUtils.publishEventPayload(securityEventTokenPayload, tenantDomain,
                             userOperationChannel.getUri());
-                }
-                else {
+                } else if (isUserCreatedFlow(event.getEventName()) && isTopicExists) {
+                    eventPayload = payloadBuilder.buildUserCreatedEvent(eventData);
+                    SecurityEventTokenPayload securityEventTokenPayload = EventHookHandlerUtils
+                            .buildSecurityEventToken(eventPayload, eventUri);
+                    EventHookHandlerUtils.publishEventPayload(securityEventTokenPayload, tenantDomain,
+                            userOperationChannel.getUri());
+                    IdentityUtil.threadLocalProperties.get().put(USER_CREATION_INITIATED, true);
```

Comment:
why this is needed ?

Reply:
Fixed aaa92d3474099d005ae78c237c76776418cb7c90